### PR TITLE
Simplify three regular expressions

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

**Review by ausi — DISMISSED** [review]
Approved.

But we can also remove the `<<<'EOD'` if we want to:

```php
        $attributeRegex = '(
            [\s/]*+                                    # Optional white space including slash
            ([^>\s/][^>\s/=]*+)                        # Attribute name
            \s*+                                       # Optional white space
            (?:=                                       # Assignment
                \s*+                                   # Optional white space
                (?|                                    # Value
                    "([^"]*)(?:"|$(*SKIP)(*FAIL))      # Double quoted value
                    |\'([^\']*)(?:\'|$(*SKIP)(*FAIL))  # Or single quoted value
                    |([^\s>]*+)                        # Or unquoted or missing value
                )                                      # Value end
            )?+                                        # Assignment is optional
        )ix';
```

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/String/HtmlAttributes.php [hunk 1]
```diff
@@ -419,34 +418,30 @@ private function escapeValue(string $name, string $value): string
...
-                    |\((?:(?R)|[^\)])*+(?:\)|$)   # ()-block
-                    |[^;{}\[\]()"']               # Anything else
-                )++
-            )ixs
-            EOD;
+        $declarationRegex = '(
+            (?:
+                \.                              # Escape
+                |"(?:\.|[^"\n])*+(?:"|\n|$)     # String token double quotes
+                |\'(?:\.|[^\'\n])*+(?:\'|\n|$)  # String token single quotes
```
Comment (leofeyer) [inline core-bundle/src/String/HtmlAttributes.php]: @ausi Is it correct to shorten `\\.` to `\.` here?

File: core-bundle/src/String/HtmlAttributes.php [hunk 2]
```diff
@@ -419,34 +418,30 @@ private function escapeValue(string $name, string $value): string
...
-            )ixs
-            EOD;
+        $propertyRegex = '(
+            ^
+            (?!\d)                               # Must not start with a digit
+            (?!-\d)                              # Must not start with a dash followed by a digit
+            -?+                                  # Optional leading dash
+            (?:
+                [a-z0-9\x80-\xFF_-]
+                |\\\\(?:[0-9a-f]{1,6}\s?|[^\n])  # Escape
```
Comment (leofeyer) [inline core-bundle/src/String/HtmlAttributes.php]: @ausi I had to convert `\\` into `\\\\` here to get rid of the `preg_match(): Unknown modifier '+'` error. Is the change correct?

**Comment by leofeyer** [comment]
I have reported the missing backtracking control verbs support to JetBrains: https://youtrack.jetbrains.com/issue/WI-73873/RegExp-check-does-not-handle-backtracking-control-verbs

File: core-bundle/src/String/HtmlAttributes.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/String/HtmlAttributes.php]:
```suggestion
                \.                              # Escape
                |"(?:\\\.|[^"\n])*+(?:"|\n|$)     # String token double quotes
                |\'(?:\\\.|[^\'\n])*+(?:\'|\n|$)  # String token single quotes
```
No, I think they have to be extendet to `\\\.` or `\\\\.`.
Did the unit tests not fail with `\.`? If so, I need to look into it.

File: core-bundle/src/String/HtmlAttributes.php (hunk 2 above)
Comment (ausi) [reply core-bundle/src/String/HtmlAttributes.php]:
```suggestion
                |\\\(?:[0-9a-f]{1,6}\s?|[^\n])  # Escape
```
The change is correct, but as our coding style is to only escape backslashes if needed, `\\\` would be the right one I think.

File: core-bundle/src/String/HtmlAttributes.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/String/HtmlAttributes.php]: > If so, I need to look into it.

Done, added test in 1133ea0df49c22ba126feb34c3c68e2f293e8075

**Review by ausi — APPROVED** [review]

**Pull request merged**